llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -448,6 +448,15 @@ Instruction *InstCombinerImpl::visitMul(BinaryOperator &I) {
   if (match(Op1, m_ZExt(m_Value(X))) && X->getType()->isIntOrIntVectorTy(1))
     return SelectInst::Create(X, Op0, ConstantInt::getNullValue(Ty));
 
+  // mul (sext X), Y -> select X, -Y, 0
+  // mul Y, (sext X) -> select X, -Y, 0
+  if (match(&I, m_c_Mul(m_OneUse(m_SExt(m_Value(X))), m_Value(Y))) &&
+      X->getType()->isIntOrIntVectorTy(1))
+    return SelectInst::Create(X,
+                              Builder.CreateNeg(Y, Y->getName() + ".neg", false,
```

Comment:
```suggestion
                              Builder.CreateNeg(Y, "", /*HasNUW=*/false,
```

I don't think the name is valuable.
BTW, `NUWNeg` is meaningless :( I will remove it later.